### ISSUE TITLE
Have a tree sitter grammar to unlock better Powershell experience in VS Code web

### ISSUE THREAD

**Issue by isidorn** [PowerShell/vscode-powershell#3714]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.

### Summary

Hi, VS Code PM here 👋 

After discussing with @TylerLeonhardt what would be the best PowerShell experience for vscode.dev and github.dev we think that the best option would be if PowerShell would contribute a Tree Sitter grammar.
With that basic [AnyCode](https://github.com/microsoft/vscode-anycode) language support would work and the experience would be much better.

I understand that adding a Tree Sitter grammar is work, but wanted to open this issue so we start a discussion and in the case that you do not have cycles to do it maybe someone from the community would help.

Thank you very much and let me know what you think

**Comment by TylerLeonhardt** [comment]
To fully enumerate the options I thought of that prompted us to open this issue:

* get PSES working in the web (involves a lot of unknowns...WASM, Blazor, but the problem here is multi-threading on the web)
* factor out the PowerShell parser which I think will be way easier to get working in WASM (still same WASM Blazor unknowns)
* Make a tree-sitter grammar for PowerShell (I know this is at least possible...but it is a lot of work)

tree-sitter has the added benefit of allowing github.com itself to light up their code navigation feature.

**Comment by SeeminglyScience** [comment]
Could have sworn @rjmholt started on one somewhere but I can't seem to find it. Am I misremembering that?

**Comment by TylerLeonhardt** [comment]
Yeah I don't think we got very far that that was useful to link here.

**Comment by rjmholt** [comment]
Repo is [here](https://github.com/PowerShell/tree-sitter-PowerShell)

**Comment by andschwa** [comment]
We're actually already looking at doing something like this but had not announced it publicly. It's an incubation project to be sure (no commitment but we're (read: I'm) quite interested).

**Comment by SydneyhSmith** [comment]
+1 to what @andschwa said, and certainly having more proposed scenarios/integration points/community interest always helps prioritization

**Comment by o-l-a-v** [comment]
Maybe this feature request could use this one?

<https://github.com/airbus-cert/tree-sitter-powershell>

The PowerShell owned one is archived:

<https://github.com/PowerShell/tree-sitter-PowerShell>
